frontend/packages/client/src/components/Community/JoinCommunityButton.js

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useState } from "react";
+import { useWebContext } from "../../contexts/Web3";
+import { useJoinCommunity, useUserRoleOnCommunity } from "../../hooks";
+
+export default function JoinCommunityButton({
+  enableJoin = false,
+  alignment = "flex-end",
+  communityId,
+  setTotalMembers = () => {},
+}) {
+  const [joinedBtnText, toggleJoinedBtnText] = useState("Joined");
+  const [borderColor, setBorderColor] = useState("transparent");
+  const [isMember, setMemberState] = useState(false);
+  const { createCommunityUser, deleteUserFromCommunity } = useJoinCommunity();
+  const { injectedProvider, user } = useWebContext();
+  const { addr } = user;
+  const memberState = useUserRoleOnCommunity({
+    addr,
+    communityId,
+    roles: ["member"],
+  });
+
+  useEffect(() => {
+    setMemberState(memberState);
+  }, [memberState]);
+
+  const handleHover = () => (isMember ? hoverToggle() : () => {});
+
+  const hoverToggle = () => {
+    if (joinedBtnText === "Joined") {
+      toggleJoinedBtnText("Leave");
+      setBorderColor("#FBD84D"); // defined in App.sass
+    } else {
+      toggleJoinedBtnText("Joined");
+      setBorderColor("transparent");
+    }
+  };
+
+  const joinCommunity = async () => {
+    const { success } = await createCommunityUser(
+      communityId,
+      user,
+      injectedProvider
+    );
+    if (success) {
+      setMemberState(true);
+      setTotalMembers((totalMembers) => ++totalMembers);
+    }
+  };
+
+  const leaveCommunity = async () => {
+    const { success } = await deleteUserFromCommunity(
+      communityId,
+      user,
+      injectedProvider
+    );
+    if (success) {
+      setMemberState(false);
+      setTotalMembers((totalMembers) => --totalMembers);
+    }
+  };
+
+  const handleMembership = () =>
+    !isMember ? joinCommunity() : leaveCommunity();
+
+  const btnText = () => (!isMember ? "Join" : joinedBtnText);
+
+  return (
+    <>
+      {enableJoin && (
```

Comment:
What are your thoughts on making the responsibility of the component displaying the button to dermine if the button should render instead of the logic being in the button component?

```js
const Parent = () => {
  return (
     {enableJoin && <JoinCommunityButton />
  )
}
```

Reply:
Yeah, that's probably a better approach, in case we have different cases come up in the future. I think I went too far in the directions of grouping it all. Will amend!